src/instruments/src/EFB/Ground/Ground.tsx

```diff
@@ -17,47 +17,171 @@
  */
 
 import React from 'react';
-import { IconCornerDownLeft, IconCornerDownRight, IconArrowDown, IconHandStop, IconTruck, IconBriefcase, IconBuildingArch, IconArchive } from '@tabler/icons'
+import { IconCornerDownLeft, IconCornerDownRight, IconArrowDown, IconHandStop, IconTruck, IconBriefcase, IconBuildingArch, IconArchive, IconHandFinger } from '@tabler/icons'
 
 import './Ground.scss'
 import fuselage from '../Assets/320neo-outline-upright.svg'
+import {setSimVar, getSimVar} from '../../util.mjs'
 
 type GroundProps = {}
 
-type GroundState = {}
+type GroundState = {
+    tugActive: boolean;
+    activeButton: string
+}
+
+let timer: number;
 
 class Ground extends React.Component<GroundProps, GroundState> {
 
+    constructor(props: GroundProps) {
+        super(props);
+
+        this.state = {
+            tugActive: false,
+            activeButton: 'none',
+        };
+    }
+
+    componentWillUnmount() {
+        clearTimeout(timer);
+    }
+
+    toggleGroundAction(action: GroundServices) {
+        setSimVar(action, "1", "boolean");
+    }
+
+    setTugHeading(action: GroundServices, direction: number) {
+            if (!this.state.tugActive) {
+                this.togglePushback(true);
+            } else {
+                const tugHeading = this.getTugHeading(direction);
+                // KEY_TUG_HEADING is an unsigned integer, so let's convert
+                setSimVar(action, (tugHeading * 11930465) & 0xffffffff, "UINT32");
+            }
+    }
+
+    getTugHeading(value: number): number {
+        const currentHeading = getSimVar("PLANE HEADING DEGREES TRUE", "degrees");
+        return (currentHeading  + value) % 360;
+    }
+
+    togglePushback(targetState: boolean) {
+        const tugActive = this.state.tugActive;
+
+        if (tugActive != targetState) {
+            setSimVar(GroundServices.TOGGLE_PUSHBACK, 1, "boolean");
+            this.setState({
+                tugActive: targetState
+            });
+        }
+    }
+
+    handleClick(callBack: () => void, event: React.MouseEvent) {
+        if (!this.state.tugActive) {
+            if (event.currentTarget.id === this.state.activeButton) {
+                this.setState({
+                    activeButton: "none"
+                });
+            } else if (!event.currentTarget.id.includes("down-")) {
+                this.setState({
+                    activeButton: event.currentTarget.id
+                });
+            }
+            callBack();
+        } else {
+            if (event.currentTarget.id.match("(down)|(stop)")) {
+                this.setState({
+                    activeButton: event.currentTarget.id
+                });
+                callBack();
+            }
+        }
+    }
```

Comment:
one issue with this code (and probably the `state.activeButton` field, is that it's possible to call more than one ground service at the same time (eg., both baggage and catering). If you do that, only the last button clicked will show as selected.